src/database/models/userArticleLink.js

```diff
@@ -0,0 +1,104 @@
+import Base from './base';
+
+/**
+ * https://g0v.hackmd.io/eIeU2g86Tfu5VnLazNfUvQ
+ */
+class UserArticleLink extends Base {
+  static get DEFAULT_DATA() {
+    return {
+      createdAt: new Date(),
+      lastViewedAt: null,
+      lastRepliedAt: null,
+      lastPositiveFeedbackRepliedAt: null,
+    };
+  }
+
+  static get collection() {
+    return 'userArticleLink';
+  }
+
+  /**
+   * @static
+   * @override
+   * @param {UserArticleLink} data
+   * @returns {UserArticleLink}
+   */
+  static async create(data) {
+    return super.create({
+      ...this.DEFAULT_DATA,
+      ...data,
+    });
+  }
+
+  /**
+   * An atomic and upsert enabled operation.
+   * @param {string} userId
+   * @param {string} articleId
+   * @returns {Promise<UserArticleLink>}
+   */
+  static async findByUserIdAndArticleId(userId, articleId) {
```

Comment:
I think it's essentially an `upsert` (mutation) operation, a bit misleading when we name it `findByXXX` here. The same goes to `UserSettings#findByUserId`. Maybe `findOrInsertByXXX` is more suitable.